app/views/fields/polymorphic/_show.html.erb

```diff
@@ -18,7 +18,7 @@ By default, the relationship is rendered as a link to the associated object.
 
 <% if field.data %>
   <%= link_to(
-    field.data.to_s,
-    polymorphic_path([:admin, field.data])
+    field.displayed_associated_resource,
```

Comment:
After seeing this in practice, I think I prefer `display_associated_resource`. There's something about `displayed` that doesn't read quite right.